fhirAssembly/src/main/resources/assembly/com.vantiq.fhir.fhirConnection/procedures/com/vantiq/fhir/fhirService_performRestOpQuery.vail

```diff
@@ -13,29 +13,53 @@ if (!theSource) {
     theSource = fhirSource.getValue()
 }
 var retVal = ""
+var headers
+if (modifiers && modifiers.headers) {
+    headers = modifiers.headers
+} else {
+    headers = {}
+}
+
+var gp
+if (modifiers && modifiers.generalParams) {
+    gp = modifiers.generalParams
+} else {
+    gp = {}
+}
 
-log.debug("Performing {} method on path {} with query {}", [method, targetPath, query])
+var comboQuery = fhirService.combineQueryAndParams(query, gp)
+
+log.debug("Performing {} method on path {} with query {} and headers {}", [method, targetPath, comboQuery, headers])
 try {
     if (method == "GET" || method == "HEAD") {
         retVal = SELECT from SOURCE @theSource with path = targetPath,
             method = method,
-            query = query,
+            query = comboQuery,
+            headers = headers,
             contentType = "application/fhir+json",
-            responseType = "application/fhir+json"
+            responseType = "application/fhir+json",
+            asFullResponse = true
```

Comment:
Use `asFullResponse` to allow the return of the numeric status code & headers.  The FHIR protocol uses those in some specific instances.